src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.netcoreapp.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    partial class Activity
+    {
+        /// <summary>
+        /// Returns high resolution (~1 usec) current UTC DateTime.
+        /// </summary>
+        internal static DateTime GetUtcNow()
+        {
+            // .NET Core CLR gives accurate UtcNow
+            return DateTime.UtcNow;
+        }
+
+        private unsafe static SpanAction<char, long> s_generateRootId = (span, rootId) =>
+        {
+            span[0] = '|';
+            char format = 'x';
+            span = span.Slice(1);
+            // It is important that the part that changes frequently be first, because
+            // many sampling functions don't sample from the high entropy part of their hash function.
+            // This makes sampling based on this produce poor samples.
+            rootId.TryFormat(span, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
```

Comment:
You can just pass "x" rather than doing this unsafe code to pass the address of a stack-based char.

Reply:
Oh, and then "x" implicitly converts to `ROS<char>`

I will admit I was a little stumped on how to pass "x" to 
```
TryFormat(Span<Char>, Int32, ReadOnlySpan<Char>, IFormatProvider) 
```
Never occurred to me to just pass "x" 😄